ProcessLib/ThermoMechanicalPhaseField/ThermoMechanicalPhaseFieldProcess-impl.h

```diff
@@ -0,0 +1,347 @@
+        dof_tables.emplace_back(*_local_to_global_index_map);
+    }
+    else
+    {
+        // For the staggered scheme
+        if (_coupled_solutions->process_id == _mechanics_related_process_id)
+        {
+            DBUG(
+                "Assemble the Jacobian equations of "
+                "temperature-deformation in "
+                "ThermoMechanicalPhaseFieldProcess for the staggered scheme.");
+        }
+
+        if (_coupled_solutions->process_id == _phase_field_process_id)
+        {
+            DBUG(
+                "Assemble the Jacobian equations of phase field in "
+                "ThermoMechanicalPhaseFieldProcess for the staggered scheme.");
+        }
+        else
+        {
+            DBUG(
+                "Assemble the Jacobian equations of "
+                "heat conduction in "
+                "ThermoMechanicalPhaseFieldProcess for the staggered scheme.");
+        }
+        dof_tables.emplace_back(
+            getDOFTableByProcessID(_heat_conduction_process_id));
+        dof_tables.emplace_back(
+            getDOFTableByProcessID(_mechanics_related_process_id));
+        dof_tables.emplace_back(
+            getDOFTableByProcessID(_phase_field_process_id));
+    }
+
+    GlobalExecutor::executeMemberDereferenced(
+        _global_assembler, &VectorMatrixAssembler::assembleWithJacobian,
+        _local_assemblers, dof_tables, t, x, xdot, dxdot_dx, dx_dx, M, K, b,
+        Jac, _coupled_solutions);
+}
+
+template <int DisplacementDim>
+void ThermoMechanicalPhaseFieldProcess<DisplacementDim>::preTimestepConcreteProcess(
+    GlobalVector const& x, double const t, double const dt,
+    const int process_id)
+{
+    DBUG("PreTimestep ThermoMechanicalPhaseFieldProcess.");
+
+    _process_data.dt = dt;
+    _process_data.t = t;
+
+    if (process_id != _mechanics_related_process_id)
+    {
+        return;
+    }
+    GlobalExecutor::executeMemberOnDereferenced(
+        &ThermoMechanicalPhaseFieldLocalAssemblerInterface::preTimestep, _local_assemblers,
+        getDOFTable(process_id), x, t, dt);
+}
+
+template <int DisplacementDim>
+void ThermoMechanicalPhaseFieldProcess<DisplacementDim>::postTimestepConcreteProcess(
+    GlobalVector const& x, int const process_id)
+{
+    DBUG("PostTimestep ThermoMechanicalPhaseFieldProcess.");
+
+    GlobalExecutor::executeMemberOnDereferenced(
+        &ThermoMechanicalPhaseFieldLocalAssemblerInterface::postTimestep, _local_assemblers,
+        getDOFTable(process_id), x);
+}
+
+template <int DisplacementDim>
+void ThermoMechanicalPhaseFieldProcess<DisplacementDim>::postNonLinearSolverConcreteProcess(
+    GlobalVector const& x, const double t, const int process_id)
+{
+    if (process_id != _mechanics_related_process_id)
+    {
+        return;
+    }
+
+    DBUG("PostNonLinearSolver ThermoMechanicalPhaseFieldProcess.");
+    // Calculate strain, stress or other internal variables of mechanics.
+    GlobalExecutor::executeMemberOnDereferenced(
```

Comment:
It can be changed as
```
const bool use_monolithic_scheme = false;
GlobalExecutor::executeMemberOnDereferenced(
         &LocalAssemblerInterface::postNonLinearSolver, _local_assemblers,
        getDOFTable(process_id), x, t, use_monolithic_scheme);
```

Reply:
changed.